diff -Naur Net-SSLeay-1.80.orig/SSLeay.xs Net-SSLeay-1.80/SSLeay.xs
--- Net-SSLeay-1.80.orig/SSLeay.xs	2017-01-04 13:39:08.000000000 -0800
+++ Net-SSLeay-1.80/SSLeay.xs	2017-03-21 17:47:41.096390399 -0700
@@ -190,7 +190,7 @@
 #ifdef OPENSSL_FIPS
 #include <openssl/fips.h>
 #endif
-#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L && !defined(OPENSSL_IS_BORINGSSL)
 #include <openssl/ocsp.h>
 #endif
 #undef BLOCK
@@ -751,7 +751,7 @@
 
 #endif
 
-#if OPENSSL_VERSION_NUMBER >= 0x10000000L && !defined(OPENSSL_NO_TLSEXT)
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L && !defined(OPENSSL_NO_TLSEXT) && !defined(OPENSSL_IS_BORINGSSL)
 
 int tlsext_status_cb_invoke(SSL *ssl, void *arg)
 {
@@ -2062,10 +2062,13 @@
 SSL_get_wbio(s)
      SSL *              s
 
+#if !defined(OPENSSL_IS_BORINGSSL)
 
 SSL_SESSION *
 SSL_SESSION_new()
 
+#endif
+
 int
 SSL_SESSION_print(fp,ses)
      BIO *              fp
@@ -4583,6 +4586,8 @@
 #endif
 #endif
 
+#if !defined(OPENSSL_IS_BORINGSSL)
+
 int
 SSL_CTX_get_ex_new_index(argl,argp,new_func,dup_func,free_func)
      long argl
@@ -4591,6 +4596,8 @@
      CRYPTO_EX_dup *   dup_func
      CRYPTO_EX_free *  free_func
 
+#endif
+
 int
 SSL_CTX_set_session_id_context(ctx,sid_ctx,sid_ctx_len)
      SSL_CTX *   ctx
@@ -4626,6 +4633,8 @@
      SSL *  ssl
      cb_ssl_int_int_ret_DH *  dh
 
+#if !defined(OPENSSL_IS_BORINGSSL)
+
 int
 SSL_get_ex_new_index(argl, argp, new_func, dup_func, free_func)
      long argl
@@ -4642,8 +4651,12 @@
      CRYPTO_EX_dup *  dup_func
      CRYPTO_EX_free * free_func
 
+#endif
+
 #define REM_SEMIAUTOMATIC_MACRO_GEN_1_09
 
+#if !defined(OPENSSL_IS_BORINGSSL)
+
 long
 SSL_clear_num_renegotiations(ssl)
   SSL *  ssl
@@ -4652,6 +4665,17 @@
   OUTPUT:
   RETVAL
 
+#endif
+
+#if defined(OPENSSL_IS_BORINGSSL)
+
+long
+SSL_CTX_add_extra_chain_cert(ctx,x509)
+     SSL_CTX *	ctx
+     X509 *     x509
+
+#else
+
 long
 SSL_CTX_add_extra_chain_cert(ctx,x509)
      SSL_CTX *	ctx
@@ -4661,6 +4685,8 @@
   OUTPUT:
   RETVAL
 
+#endif
+
 void *
 SSL_CTX_get_app_data(ctx)
      SSL_CTX *	ctx
@@ -4669,6 +4695,14 @@
   OUTPUT:
   RETVAL
 
+#if defined(OPENSSL_IS_BORINGSSL)
+
+long
+SSL_CTX_get_mode(ctx)
+     SSL_CTX *	ctx
+
+#else
+
 long
 SSL_CTX_get_mode(ctx)
      SSL_CTX *	ctx
@@ -4677,6 +4711,16 @@
   OUTPUT:
   RETVAL
 
+#endif
+
+#if defined(OPENSSL_IS_BORINGSSL)
+
+long
+SSL_CTX_get_read_ahead(ctx)
+     SSL_CTX *	ctx
+
+#else
+
 long
 SSL_CTX_get_read_ahead(ctx)
      SSL_CTX *	ctx
@@ -4685,6 +4729,16 @@
   OUTPUT:
   RETVAL
 
+#endif
+
+#if defined(OPENSSL_IS_BORINGSSL)
+
+long
+SSL_CTX_get_session_cache_mode(ctx)
+     SSL_CTX *	ctx
+
+#else
+
 long
 SSL_CTX_get_session_cache_mode(ctx)
      SSL_CTX *	ctx
@@ -4693,6 +4747,8 @@
   OUTPUT:
   RETVAL
 
+#endif
+
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
 long
 SSL_CTX_need_tmp_RSA(ctx)
@@ -5839,7 +5895,7 @@
 
 #endif
 
-#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L && !defined(OPENSSL_IS_BORINGSSL)
 
 #if !defined(OPENSSL_NO_TLSEXT)
 
diff -Naur Net-SSLeay-1.80.orig/inc/Module/Install/PRIVATE/Net/SSLeay.pm Net-SSLeay-1.80/inc/Module/Install/PRIVATE/Net/SSLeay.pm
--- Net-SSLeay-1.80.orig/inc/Module/Install/PRIVATE/Net/SSLeay.pm	2017-01-04 13:39:55.000000000 -0800
+++ Net-SSLeay-1.80/inc/Module/Install/PRIVATE/Net/SSLeay.pm	2017-03-21 17:11:52.867951191 -0700
@@ -24,20 +24,7 @@
     $self->requires_external_cc;
 
     my $prefix = $self->find_openssl_prefix;
-    my $exec   = $self->find_openssl_exec($prefix);
-
-    unless (-x $exec) {
-        print <<EOM;
-*** Could not find OpenSSL
-    If it's already installed, please set the OPENSSL_PREFIX environment
-    variable accordingly. If it isn't installed yet, get the latest version
-    from http://www.openssl.org/.
-EOM
-        exit 0; # according http://wiki.cpantesters.org/wiki/CPANAuthorNotes this is best-practice when "missing library"
-    }
-
-    $self->check_openssl_version($prefix, $exec);
-    my $opts = $self->ssleay_get_build_opts($prefix, $exec);
+    my $opts = $self->ssleay_get_build_opts($prefix);
 
     $self->makemaker_args(
         CCCDLFLAGS => $opts->{cccdlflags},
@@ -58,7 +45,7 @@
 }
 
 sub ssleay_get_build_opts {
-    my ($self, $prefix, $exec) = @_;
+    my ($self, $prefix) = @_;
 
     my $opts = {
         lib_links  => [],
@@ -209,64 +196,6 @@
     return;
 }
 
-sub find_openssl_exec {
-    my ($self, $prefix) = @_;
-
-    my $exe_path;
-    for my $subdir (qw( bin sbin out32dll ia64_exe alpha_exe )) {
-        my $path = File::Spec->catfile($prefix, $subdir, "openssl$Config{_exe}");
-        if ( -x $path ) {
-            return $path;
-        }
-    }
-    ($prefix) = $self->check_no_path
-       and return $prefix;
-    return;
-}
-
-sub check_openssl_version {
-    my ($self, $prefix, $exec) = @_;
-    my ($major, $minor, $letter);
-
-    {
-        my $pipe = gensym();
-        open($pipe, "$exec version |")
-            or die "Could not execute $exec";
-        my $output = <$pipe>;
-        chomp $output;
-        close $pipe;
-
-	if ( ($major, $minor, $letter) = $output =~ /^OpenSSL\s+(\d+\.\d+)\.(\d+)([a-z]?)/ ) {
-	    print "*** Found OpenSSL-${major}.${minor}${letter} installed in $prefix\n";
-	} elsif ( ($major, $minor) = $output =~ /^LibreSSL\s+(\d+\.\d+)/ ) {
-	    print "*** Found LibreSSL-${major}.${minor} installed in $prefix\n";
-	} else {
-            die <<EOM
-*** OpenSSL version test failed
-    (`$output' has been returned)
-    Either you have bogus OpenSSL or a new version has changed the version
-    number format. Please inform the authors!
-EOM
-        }
-    }
-
-    if ($major < 0.9 || ($major == 0.9 && $minor < 3)) {
-        print <<EOM;
-*** That's too old!
-    Please upgrade OpenSSL to the latest version (http://www.openssl.org/)
-EOM
-        exit 0; # according http://wiki.cpantesters.org/wiki/CPANAuthorNotes this is best-practice when "missing library"
-    }
-
-    if ($major == 1.1 && $minor > 0) {
-        print <<EOM;
-*** That's newer than what this module was tested with
-    You should consider checking if there is a newer release of this module
-    available. Everything will probably work OK, though.
-EOM
-    }
-}
-
 sub fixpath {
     my ($self, $text) = @_;
 
